src/systems/filecoin_vm/runtime/impl/runtime.go

```diff
@@ -660,9 +662,10 @@ func (rt *VMContext) GetRandomness(epoch abi.ChainEpoch) abi.RandomnessSeed {
 func (rt *VMContext) NewActorAddress() addr.Address {
 	addrBuf := new(bytes.Buffer)
 
-	err := rt._immediateCaller.MarshalCBOR(addrBuf)
+	// note that this is a pubkey address
+	err := rt._toplevelMessage.From.MarshalCBOR(addrBuf)
```

Comment:
No, there's no guarantee at all that `From` is a pubkey address. In fact, senders pay less gas if it's an ID address.

This needs to be resolved through the sender account actor state. That could be done inside this object, or before the ctx is constructed (pass it as a new ctor param).